### ISSUE TITLE
Add an option to disable the adulterated CTCP VERSION response

### ISSUE THREAD

**Issue by wodim** [znc/znc#1012]
Lately I have noticed znc adds “via ZNC - http://znc.in” to my CTCP VERSION responses. I would like to have an option to disable such behaviour.

**Comment by t3chguy** [comment]
Maybe making the suffix configurable would be better behavior as then you could add a custom Suffix which would default to the `via ZNC - http://znc.in`

**Comment by wodim** [comment]
@t3chguy I don’t see how a configurable suffix would be of use to any user, but that would be a good idea as well.

**Comment by t3chguy** [comment]
Could be useful if you did want to keep the ZNC Suffix but wanted to make it coloured or something

**Comment by t3chguy** [comment]
https://github.com/znc/znc/blob/d39b20b505e1b44f5d565fc61ce5cf3fe2ca6283/src/znc.cpp#L118-L131

**Comment by DarthGandalf** [comment]
You can put your wanted version reply to "CTCP replies", as reply to VERSION. Then "via ZNC" won't be appended.

**Comment by t3chguy** [comment]
But then the CTCP Request will not be passed through to the connected clients

**Comment by wodim** [comment]
@DarthGandalf this is not the resolution I expect. I don't want to have to hardcode a reply. I use several clients, sometimes connected at once.

**Comment by DarthGandalf** [comment]
@t3chguy Right, it won't. ZNC will reply itself with the configured reply.

**Comment by DarthGandalf** [comment]
@wodim So you want to show actual version of your clients, but not of ZNC?

**Comment by wodim** [comment]
@DarthGandalf exactly. I expect ZNC to sit in between the server and my clients, proxying the traffic, but not messing with it unless I explicitly tell it to do so.

**Comment by dgw** [comment]
I expect the same behavior as @wodim

**Comment by DarthGandalf** [comment]
You can copypaste actual client version to that field then.
Alternatively, you can write a module which hooks `OnUserCTCPReply` and removes the "via" part.

**Comment by t3chguy** [comment]
Your first solution wouldn't work with multiple connected clients @DarthGandalf

**Comment by wodim** [comment]
I am not happy with the resolution, I guess I have hit a wall of chauvinism.

**Comment by DarthGandalf** [comment]
Second solution works though.

**Comment by DarthGandalf** [comment]
Ok, let's close this issue when someone writes such module and puts link to wiki as an external.

**Comment by pythonmcpi** [comment]
I know this is an old issue but it doesn't seem to be fixed in 2020... I'm going to try to write a plugin for this

**Comment by DarthGandalf** [comment]
> I know this is an old issue but it doesn't seem to be fixed in 2020...

Perhaps everyone is satisfied with the solution of using the CTCP response defined by user?

> I'm going to try to write a plugin for this

Okay. Come to #znc-dev if you need help with it

**Comment by wodim** [comment]
Or maybe those who are bothered by it are not bothered enough to write a plugin. (I count myself there)

**Comment by pythonmcpi** [comment]
https://github.com/pythonmcpi/CustomCTCPVersion.py
Requires modpython
